api/src/services/user.service.ts

```diff
@@ -16,7 +16,7 @@ export class UserService {
     @InjectRepository(ProjectUser) private projectUsersRepo: Repository<ProjectUser>,
   ) {}
 
-  async create(name: string, email: string, password: string): Promise<User> {
+  async create({ grantType, email, name, password }: { grantType: GrantType; email: string; name: string; password?: string }): Promise<User> {
     const normalizedEmail = this.normalizeEmail(email);
     const exists = await this.userRepo.findOne({ email: normalizedEmail });
```

Comment:
A common behaviour is that if a user tries to sign up again with a provider (instead of logging in), it would not fail as if the user record already existed but instead just auto login that user. I think we should probably add that check here.

Reply:
Ok, I blocked it intentionally because I didn't know what exactly you wanted but I will change it

Reply:
What do you want exactly? Because now I only block the signup but if you login with an old account that will work. Do you want the same gesture for login and signup? And if you try to login without an account I create one?

Reply:
Yes, basically the only addition is that when a user clicks on "Sign up" (as in register) even though he already had an account with the same credentials, we still accept that and just log him provided that the external provided verified the credentials.

Reply:
Hang on I will provide a branch with some suggestions so we can try it together